impl/src/main/java/org/opendaylight/unimgr/mef/nrp/common/NrpDao.java

```diff
@@ -79,21 +82,31 @@ public Node createNode(String topologyId, String nodeId, LayerProtocolName name,
     }
 
     public Node createNode(String topologyId, String nodeId, String activationDriverId, LayerProtocolName name, List<OwnedNodeEdgePoint> neps) {
+        return createNode(topologyId, nodeId, activationDriverId, name, neps,null);
+    }
+
+    public Node createNode(String topologyId, String nodeId, String activationDriverId, LayerProtocolName name, List<OwnedNodeEdgePoint> neps, List<ServiceVlanMap> serviceVlanMapList) {
         verifyTx();
         assert tx != null;
         Uuid uuid = new Uuid(nodeId);
 
-        Node node = new NodeBuilder()
+        NodeBuilder nb = new NodeBuilder()
                 .setKey(new NodeKey(uuid))
                 .setUuid(uuid)
                 .setLayerProtocolName(Collections.singletonList(name))
                 .setOwnedNodeEdgePoint(neps)
-                .addAugmentation(Node1.class, new Node1Builder().setActivationDriverId(activationDriverId).build())
-                .build();
-        tx.put(LogicalDatastoreType.OPERATIONAL, node(new Uuid(topologyId), new Uuid(nodeId)), node);
+                .addAugmentation(Node1.class, new Node1Builder().setActivationDriverId(activationDriverId).build());
+
+        Node node = serviceVlanMapList == null ? nb.build() : nb.addAugmentation(Node2.class, new Node2Builder().setServiceVlanMap(serviceVlanMapList).build()).build();
+        tx.put(LogicalDatastoreType.OPERATIONAL, node(nodeId), node);
         return node;
     }
 
+    public void updateNode(Node node) {
```

Comment:
this is in fact an override of a node, how about renaming operation?
please bare in mind that all external changes between reading/modyfying the node given as parameter and writing it are silently lost

Reply:
it was modeled on method updateNep.
the name can be changed. Is 'putNode' ok ?